mount/mount.go

```diff
@@ -27,6 +27,8 @@ type Mount struct {
 	// Options contains zero or more fstab-style mount options. Typically,
 	// these are platform specific.
 	Options []string
+	// ExtraOptions contains extra options except fstab-style mount options.
+	ExtraOptions []string
```

Comment:
How is this used for mounting a filesystem?
If the purpose is configuring the runtime, isn't task-related structs (e.g. `TaskInfo`) the proper place to fix?

Reply:
@ktock This field is not used to mount a filesystem like overlayfs, it's main role is used to delivery config information. As remote-snapshotter(take [stargz-snapshotter](https://github.com/containerd/stargz-snapshotter/blob/main/snapshot/snapshot.go#L288) for example) only return `[]mount.Mount`, so if we want to delivery some config information from remote-snapshotter to runtime, I think adding `ExtraOptions` field is a suitable way.

In containerd, it will ignore `ExtraOptions` and do mount overlayfs as original logic. In runtime, it will parse the `ExtraOptions` and use it. In nydus case, nydus-snapshotter will add nydus image metedata file path and config to `ExtraOptions` and delivery to kata-runtime, kata-runtime used these info to request nydusd (fuse/virtiofs daemon) to 
mount nydus rafs fllesystem.

Reply:
Wondering if this should be more of a `MetaData` field (making it more clear it's entirely up to the consumer what to use it for)

Reply:
@thaJeztah Do you mean renaming `ExtraOption []string` to `MetaData []string`? It does make it more clear.

Reply:
yes, I was thinking along those lines (but of course could use input from others); mostly was reading along, and if I understand it correctly, effectively this information is arbitrary "extra information", that _can_ be used by other systems for `<insert purpose here>`.

Naming it MetaData (or something like that), makes it clearer that whatever is in that field is not a strict "contract" (from containerd's perspective).

Of course the description of the field should also outline this, and could (perhaps) add some recommendations (e.g. recommend to use a common prefix (`io.katacontainers.opt=foo`)) <--- thinking out loud here; I know we did the same recommendation for labels

Reply:
@thaJeztah Thanks for you advise, I agree that `MetaDatas []string` is more clear. I have updated the field and it's description, hope for your reivew again.